### ISSUE TITLE
(Style of select/multiple) 把select标签单选的样式用到multiple="multiple"多选上，会更符合思维

### ISSUE THREAD

**Issue by tianyao1111** [w3c/html#1291]
我最近碰见一个问题，用select标签实现多选，发现multiple="multiple"的样式展开很难看，而select单选的样式是我心目中多选的样子，为什么不让多选样式也可以折叠呢？

**Comment by Dafrok** [comment]
The W3C standard describes:

> The multiple attribute is a boolean attribute. If the attribute is present, then the select element represents a control for selecting zero or more options from the list of options. If the attribute is absent, then the select element represents a control for selecting a single option from the list of options.

> If the multiple attribute is absent and the element's display size is greater than 1, then the user agent should also allow the user to request that the option whose selectedness is true, if any, be unselected. Upon this request being conveyed to the user agent, and before the relevant user interaction event is queued (e.g. before the click event), the user agent must set the selectedness of that option element to false and then queue a task to fire a simple event that bubbles named change at the select element, using the user interaction task source as the task source.

> If the multiple attribute is absent and the element's display size is 1, then whenever there are no option elements in the select element's list of options that have their selectedness set to true, the user agent must set the selectedness of the first option element in the list of options in tree order that is not disabled, if any, to true.

> If the multiple attribute is present, and the element is not disabled, then the user agent should allow the user to toggle the selectedness of the option elements in its list of options that are themselves not disabled (either through a click, or through a menu command, or any other mechanism). Upon the selectedness of one or more option elements being changed by the user, and before the relevant user interaction event is queued (e.g. before a related click event), the user agent must queue a task to fire a simple event that bubbles named change at the select element, using the user interaction task source as the task source.

There is no specific of the style of the native SELECT element, it's implemented by the browsers themselves and has nothing to do with the standard.

**Comment by chaals** [comment]
@tianyao1111 @Dafrok thank you for the issue and comments.

It is true that the style is implemented by the browser. But if I understand the issue correctly (using automatic translation :S ), it is asking if there could be better guidance on how the multiple select is presented.

It makes sense to me to *suggest* that  browsers present elements in a way that is easy to understand, and we have a rendering chapter for that. (The rendering is mostly not required for implementation, jjust suggested).

I did not understand the specific proposal, but the question is definitely appropriate, and it might make sense to change the specification to suggest specific rendering, or at least some presentation characteristics to reduce confusion for users. (I would appreciate browsers doing that better, as a single user...)

**Comment by chaals** [comment]
(If I got the edited title wrong, please feel free to suggest an improvement).

**Comment by Dafrok** [comment]
As @tianyao1111's description is not very clear, I guess he wants to said that the SELECT element should fold its options when `size` is 1 and `multiple` is true, just performance like a default SELECT element. 
I'm not sure if this feature should belong to the standard.

**Comment by siusin** [comment]
I think @tianyao1111 is suggesting the browsers to provide a similar UI for `<select>` with `multiple` attr  set (i.e. `<select multiple>`) and without `multiple` attr set...

I agreed with @Dafrok that this might be out of the scope of the HTML spec.

**Comment by chaals** [comment]
If the issue is specific to `size="1"` - which is effectively the same as a non-multiple, I agree that it would be clearer to users to make it behave the same way. But for example it may bbe the case that the size is variable through the life of the page, and signalling that it is at least in principle a `multiple` select is helpful.

So I think this is outside the scope - we suggest basic rendering guidelines, but I think this belongs in the area of implementations being free to choose such details of UI.

Closing.

Thank you @dafrok and @siusin for explanations.
